### PR TITLE
Refactoring

### LINKED ISSUE #568: Moving utils.pull to BaseDataProvider
# Feature request

## Thesis
I suggest moving `utils.pull` to `BaseDataProvider` because it's only needed for locale-dependent providers, which are all subclasses of `BaseDataProvider`.

## Reasoning
We are forced to always manually import `pull` to the modules where are we create new providers and repeat the same code, although the pull is needed only for locale-dependent providers which are all inherited from `BaseDataProvider`. It will be much logically correct to keep pull in provider `BaseDataProvider` I think.

The idea is simple, instead of this:
```python
from mimesis.utils import pull

class OldProvider(BaseDataProvider):

    def __init__(self, *args, **kwargs):
        """Initialize attributes.
        """
        super().__init__(*args, **kwargs)
        self._datafile = 'datetime.json'
        self._data = pull(self._datafile, self.locale)
```

Use this:

```python
class NewProvider(BaseDataProvider):

    def __init__(self, *args, **kwargs):
        """Initialize attributes.
        """
        super().__init__(*args, **kwargs)
        self._datafile = 'datetime.json'
        self.pull(self._datafile)
```
After this we'll able to use pull withot any extra importing.

That's just discussion, I have not started working on this yet.

### LINKED ISSUE #569: Merge modules config and locales
# Feature request

## Thesis
I suggest merging this two modules (`config.py` and `locales.py`). The module `confing.py` contains only things related to locale, so it's better to keep only module `locales.py` and move all the content of `config.py` to `locales.py`.

## Reasoning

Here is no need in two different modules for the same things.

### PR DESCRIPTION
**Pull request by lk-geimfari**
# I have made things!

## Checklist

- [x] Merge `config.py` and` locales.py`
- [x] Move pull to `BaseDataProvider`
- [x] Update locales.override to support a new way of data pulling from files.
- [x] Delete module `utils.py`
- [x] Rewrite tests for moved functions

## Related issues

- Closes #568, #569, #570

### CODE REVIEW

**Comment by lk-geimfari** [comment]
I know that' really hard to review such PRs, but there is no another way.

**Comment by lk-geimfari** [comment]
@sobolevn I need your review again and after this time I'll merge this PR if you have not any pieces of advice/notices.

**Comment by lk-geimfari** [comment]
I did not fix this https://github.com/lk-geimfari/mimesis/issues/579, because I'm not sure how to do it.

File: mimesis/providers/base.py [hunk 1]
```diff
@@ -71,16 +71,16 @@ def __str__(self) -> str:
...
+                 seed: Seed = None) -> None:
         """Initialize attributes for data providers.
 
         :param locale: Current locale.
         :param seed: Seed to all the random functions.
         """
         super().__init__(seed=seed)
         self._data: dict = {}
-        self._datafile: Optional[str] = None
+        self._datafile: str = ''
```
Comment (sobolevn) [inline mimesis/providers/base.py]: Useless annotation

File: mimesis/providers/base.py [hunk 2]
```diff
@@ -71,16 +71,16 @@ def __str__(self) -> str:
...
-                 seed: Optional[Seed] = None) -> None:
+    def __init__(self, locale: str = locales.DEFAULT_LOCALE,
+                 seed: Seed = None) -> None:
         """Initialize attributes for data providers.
 
         :param locale: Current locale.
         :param seed: Seed to all the random functions.
         """
         super().__init__(seed=seed)
         self._data: dict = {}
```
Comment (sobolevn) [inline mimesis/providers/base.py]: Should be annotated as: `Dict[KeyType, ValueType`. `dict` means `Dict[Any, Any]`. And it does not help much.

Comment (lk-geimfari) [reply mimesis/providers/base.py]: @sobolevn The problem is that we don't know `ValueType`.

Comment (sobolevn) [reply mimesis/providers/base.py]: It may be `Any` in this case.

Comment (sobolevn) [reply mimesis/providers/base.py]: It might be useful to enable `disallow_any_generics = True`

**Pull request closed without merging**